quinn/src/endpoint.rs

```diff
@@ -228,9 +242,20 @@ impl EndpointInner {
     }
 
     fn drive_incoming(&mut self) {
-        for conn in &mut self.incoming {
-            // It's safe to poll an already dead connection
-            let _ = conn.poll();
+        if let Some(ref mut incoming) = self.incoming {
```

Comment:
Maybe hoist the `if` out of this method so that the `else` part can go in a different method? As is this seems pretty confusing and conceptually wrong (`drive_incoming` has nothing to do with the `draining` loop).

Reply:
`draining` is, strictly speaking, still a set of incoming connections, just ones that we've determined the application will no longer receive. Maybe it just needs a better name? It's nice to have the if statement directly enclosing the loop because then we don't need any unwraps or duplicate checks.

Reply:
Hmm, okay. I can't think of a better name, maybe just add a short explanatory comment to explain what's going on here, why it's doing what it's doing? (Also, too bad `VecDeque::retain()` doesn't give you `&mut T` rather than `&T` -- maybe I'll file a bug about that since I don't see any downside.)

Reply:
Renamed to `orphaned_incoming`, which strikes me as a suitable improvement.

See [discussion on `Vec::retain`](https://github.com/rust-lang/rust/issues/25477) for why we don't have a `&mut` version; apparently [`drain_filter`](https://github.com/rust-lang/rust/issues/43244) will subsume that wart eventually.

Reply:
I just realized that this is all algebraically equivalent to having a single container and a boolean flag on the side, which is simpler. Refactored into that form.